jumanji/environments/routing/op/env.py

```diff
@@ -0,0 +1,313 @@
+from typing import Optional, Sequence, Tuple
+
+import chex
+import jax
+import jax.numpy as jnp
+import matplotlib
+from chex import PRNGKey
+from numpy.typing import NDArray
+
+
+import jumanji
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.types import TimeStep, restart, termination, transition
+from jumanji.viewer import Viewer
+import generator, types_, reward, constants
```

Comment:
Same remark here concerning the absolute import. Also, you shouldn't have to call your file "types_". If imports are correctly handled, you should be able to call it "types".